### PR TITLE
quast 5.0.0

### PR DESCRIPTION
**Pull request by alexeigurevich**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by alexeigurevich** [comment]
Hi, Shaun @sjackman !

Could you assist with updating Quast to the new major release?
I see that the problem in the failed Linux builds is
`/usr/bin/env: ‘python’: No such file or directory`
when the system tries to execute `./quast.py --test` (and quast.py indeed starts with `#!/usr/bin/env python`).
However, I do not understand how this should be properly fixed. Note that my formula is based on existing (and working) quast-4.6.3 formula and `./quast.py` file always had `#!/usr/bin/env python` in the beginning.

Thanks,
Alexey

**Comment by sjackman** [comment]
Hi, Alexei! Yeah, I've been waiting the release of QUAST 5 with anticipation!

Let's use Python 3 rather than Python 2. Please add `depends_on "python"` to depend on Python 3 (currently Python 3.7), and change the shebang line to `#!/usr/bin/env python3`.

macOS is using the host's `/usr/bin/python` (2.7.10), which is why it succeeds. On Linux, we don't assume the host has any Python interpreter installed.

Thanks for this PR!

**Comment by alexeigurevich** [comment]
Ok, got it. Is it possible to specify some "patch" to be applied after unpacking Quast `.tar.gz` archive? Or should we release something like 5.0.1 with the modified shebang? We can't change anything in already uploaded 5.0.0.tar.gz. 

Moreover, I am not sure that releasing a general `.tar.gz` with explicitly specified `python3` is a perfect solution. Since a user may have python2 only, too. Do you know whether there is a universal solution?

**Comment by sjackman** [comment]
You can patch the `quast` executable after installation. Add…
```ruby
inreplace bin/"quast", "#!/usr/bin/env python", "#!/usr/bin/env python3
```

**Comment by alexeigurevich** [comment]
I slightly modified your suggestion (otherwise it was not working due to symlinking issue). And also specified this `inreplace` for Linux only. Please take a look whether you agree with my version.

**Comment by sjackman** [comment]
I'd prefer to use Python 3 on both macOS and Linux, for consistency on both platforms. Sound good?

File: Formula/quast.rb [hunk 1]
```diff
@@ -14,20 +15,20 @@ class Quast < Formula
...
-    # Remove the precompiled E-MEM binary.
-    rm_f "quast_libs/MUMmer/e-mem-osx"
     prefix.install Dir["*"]
-    bin.install_symlink "../quast.py", "../metaquast.py", "../icarus.py",
-      "quast.py" => "quast", "metaquast.py" => "metaquast", "icarus.py" => "icarus"
+    bin.install_symlink "../quast.py", "../metaquast.py", "../quast-lg.py", "../icarus.py",
+      "quast.py" => "quast", "metaquast.py" => "metaquast", "quast-lg.py" => "quast-lg", "icarus.py" => "icarus"
     # Compile the bundled aligner so that `brew test quast` does not fail.
     system "#{bin}/quast", "--test"
+    if OS.linux?
```
Comment (sjackman) [inline Formula/quast.rb]: Please move the `inreplace` above `system "#{bin}/quast", "--test"` and add a blank line before and after.

Comment (alexeigurevich) [reply Formula/quast.rb]: Yes, sure. Sorry, I am testing `brew install --build-from-source` on Mac, so it is difficult to catch everything.

Comment (sjackman) [reply Formula/quast.rb]: No worries at all, Alexei.

**Comment by alexeigurevich** [comment]
>I'd prefer to use Python 3 on both macOS and Linux, for consistency on both platforms. Sound good?

Yes, that makes sense. However, I tried to specify 
`  depends_on "python3"` in the formula for consistency but `brew audit` told me that this is not good and I should use simply `"python"` instead (which probably means python2 on Mac). Should I ignore `brew audit`?

**Comment by sjackman** [comment]
`depends_on "python"` is Python 3 on both Linux and macOS. On macOS if you want to use the host's Python interpreter, you don't add any dependency at all. The dependency on the host OS is implied.

**Comment by sjackman** [comment]
Is `quast-lg` a synonym for `quast --large`?

**Comment by sjackman** [comment]
`brew postinstall python` failed on TravisCI, but that's unrelated to this PR, and not critical.
https://travis-ci.org/brewsci/homebrew-bio/builds/414264629#L3210

**Comment by sjackman** [comment]
Merged! Thanks, Alexei!

**Comment by alexeigurevich** [comment]
> Is quast-lg a synonym for quast --large?
Yes, exactly. We added it just in case and mostly for new users who will start familiarize themselves with Quast from Quast-LG.

**Comment by sjackman** [comment]
No worries at all! My pleasure.

**Pull request closed without merging**